shared/src/main/scala/kinesis4cats/consumer/Record.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2023-2023 etspaceman
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kinesis4cats
+package consumer
+
+import scala.util.Success
+import scala.util.Try
+
+import java.time.Instant
+
+import cats.syntax.all._
+import scodec.bits.ByteVector
+
+import kinesis4cats.models.EncryptionType
+import kinesis4cats.protobuf.messages
+
+final case class Record(
+    sequenceNumber: String,
+    approximateArrivalTimestamp: Instant,
+    data: ByteVector,
```

Comment:
perhaps this should be `Array[Byte]` to match the producer's `Record` class. But ByteVector seems to be safer.

Reply:
Is there any major difference? Other than arrays being mutable

Reply:
Safer API for the users yeah.